modules/frontend/pipeline/pipeline.go

```diff
@@ -65,6 +70,18 @@ func (r *HTTPRequest) ResponseData() any {
 	return r.responseData
 }
 
+func (r *HTTPRequest) Weight() int {
+	return r.weight
+}
+
+func (r *HTTPRequest) SetWeight(w int) {
+	r.weight = w
+}
+
+func (r *HTTPRequest) FromHTTPRequest(request *http.Request) *HTTPRequest {
```

Comment:
i think we can remove this method and use the Clone method (once we add the weight).

see my comment in the metrics_query_range_sharder.go file

Reply:
finishing the review i see it's used more below. if we do need this i think we should call it:
```
CloneWithHTTPRequest(request *http.Request)
```
and have it copy all internal fields except for the http request which it gets from the arg.

Reply:
Is there a clone method for Pipeline HTTPRequest? The one I see is for the  stdlib http.Request

Reply:
apologies, see above. i could have sworn there was, but i'm wrong.